microsoft/testsuites/vm_extensions/azure_disk_encryption.py

```diff
@@ -0,0 +1,193 @@
+    application_id = runbook.service_principal_client_id
+    object_id = get_identity_id(platform=platform, application_id=application_id)
+    if object_id is None:
+        raise ValueError("Environment variable 'object_id' is not set.")
+
+    node_capability = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)
+    location = node_capability.location
+    shared_resource_group = runbook.shared_resource_group_name
+
+    # Create key vault if there is not available adelisa key vault for that region
+    existing_vault = get_matching_key_vault_name(
+        platform, location, shared_resource_group, r"adelisa-\w{5}"
+    )
+    if existing_vault:
+        vault_name = existing_vault
+    else:
+        random_str = generate_random_chars(string.ascii_lowercase + string.digits, 5)
+        vault_name = f"adelisa-{random_str}"
+
+    vault_properties = VaultProperties(
+        tenant_id=tenant_id,
+        sku=KeyVaultSku(name="standard"),
+        enabled_for_disk_encryption=True,
+        access_policies=[
+            AccessPolicyEntry(
+                tenant_id=tenant_id,
+                object_id=object_id,
+                permissions=Permissions(
+                    keys=["all"], secrets=["all"], certificates=["all"]
+                ),
+            ),
+        ],
+    )
+    # If the KV exists, this will just ensure the properties are correct for ADE
+    keyvault_result = create_keyvault(
+        platform=platform,
+        location=location,
+        vault_name=vault_name,
+        resource_group_name=shared_resource_group,
+        vault_properties=vault_properties,
+    )
+
+    # Check if KeyVault is successfully created or updated before proceeding
+    assert (
+        keyvault_result
+    ), f"Failed to create or update KeyVault with name: {vault_name}"
+
+    # Run ADE Extension
+    extension_name = "AzureDiskEncryptionForLinux"
+    extension_publisher = "Microsoft.Azure.Security"
+    extension_version = "1.4"
+    settings = {
+        "EncryptionOperation": "EnableEncryption",
+        "KeyVaultURL": keyvault_result.properties.vault_uri,
+        "KeyVaultResourceId": keyvault_result.id,
+        "KeyEncryptionAlgorithm": "RSA-OAEP",
+        "VolumeType": "Os",
+    }
+    extension = node.features[AzureExtension]
+    extension_result = extension.create_or_update(
+        name=extension_name,
+        publisher=extension_publisher,
+        type_=extension_name,
+        type_handler_version=extension_version,
+        settings=settings,
+    )
+    return extension_result
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="Tests for the Azure Disk Encryption (ADE) extension",
+    requirement=simple_requirement(
+        min_memory_mb=8 * 1024,
+        supported_features=[AzureExtension],
+        supported_platform_type=[AZURE],
+        supported_os=[Ubuntu, CBLMariner, CentOs, Oracle, Redhat],
```

Comment:
@squirrelsc / @LiliDeng my test seems to be failing now after the change adding min_memory to simple_requirements.

I am seeing my test skipped for unsupported OS when using a supported OS. I tried Mariner, RHEL, and Ubuntu. Here is the output and last command:

![image](https://github.com/microsoft/lisa/assets/36088030/0328a564-f29b-4e49-84a2-d2b39dc3234e)

Reply:
Can you check os matrix like other extensions? They use a distro list, and use isinstance to check.

Reply:
If I am looking at the right setup (azureperformancediagnostics.py), this means we would have to come in here and manually update this support any time we make a support addition in ADE, or any time a new image comes out? Would be cool if we could set a minimum supported OS Major version, so that for example when RHEL 10 comes out, we do not all have to modify the code to allow RHEL 10 testing. 

It seems at least for minor versions this will work well for our case, as we have some older major versions we don't support. But just a thought for future improvement if the supported_os feature is not working.

Reply:
Update pushed. Did some additional refactoring to clean up my helper methods

Reply:
If you can support latest versions without change the code, that would be great. You can modify the logic to have minimal versions like `os.version >= "8.0.0"`